### PR TITLE
Make Apollo client configurable from the plugin system

### PR DESCRIPTION
**Pull request by blakewilson**
## Description

This PR abstracts the Apollo client from being specified in the app's project to being configurable from the plugin system.

### CODE REVIEW

File: examples/next/faust-nx/pages/_app.tsx [hunk 1]
```diff
@@ -1,12 +1,11 @@
 import 'faustnx.config';
 import React from 'react';
-import client from 'client';
 import { FaustNXProvider } from 'faust-nx';
 import type { AppProps } from 'next/app';
 
 export default function MyApp({ Component, pageProps }: AppProps) {
   return (
-    <FaustNXProvider client={client} pageProps={pageProps}>
```
Comment (theodesp) [inline examples/next/faust-nx/pages/_app.tsx:9]: I think we should still allow the user to supply their own client here irrespectively of the plugin system.

Comment (blakewilson) [reply examples/next/faust-nx/pages/_app.tsx:9]: With the `apolloClientOptions` filter hook, they could still supply their own client that operates within the bounds of the plugin system, e.g: 

```ts
class FaustNXApolloClient {
  apply({addFilter}) {
    addFilter('apolloClientOptions', 'faust-nx', (options, ctx) => {
      return {
        ssrMode: typeof window === 'undefined',
        connectToDevTools: typeof window !== 'undefined',
        link: new HttpLink({
          uri: process.env.NEXT_PUBLIC_GRAPHQL_ENDPOINT,
          fetchOptions: {
            method: 'POST',
          },
        }),
        cache: new InMemoryCache({
          possibleTypes,
          typePolicies: {
            RootQuery: {
              queryType: true,
            },
            RootMutation: {
              mutationType: true,
            },
          },
        }),
      }
    })
  }
}
```

Do you think this is a suitable altnerative?

Comment (theodesp) [reply examples/next/faust-nx/pages/_app.tsx:9]: Sure thats fine

**Review by theodesp — APPROVED** [review]
Thanks!

**Pull request merged**